sweep/sweeper.go

```diff
@@ -222,6 +222,30 @@ func (p *SweeperInput) terminated() bool {
 	}
 }
 
+// isMature returns a boolean indicating whether the input has a timelock that
+// has been reached or not. The locktime found is also returned.
+func (p *SweeperInput) isMature(currentHeight uint32) (bool, uint32) {
+	locktime, _ := p.RequiredLockTime()
+	if currentHeight < locktime {
```

Comment:
Q: Maybe a bit offtopic, but if no locktime is required will the final sweep tx have at least a locktime of the currentBlockHeight, I read somewhere is kind of beneficial to prevent miners from some reordering of transactions iirc ?

Reply:
yeah it prevents fee sniping - i think we always set that field when creating the tx (supposed there's no CLTV).